File: agora-easycall-ios-oc/AgoraSample/Menu/AGSProfileViewController.m
//
//  AGSProfileViewController.m
//  AgoraSample
//
//  Created by Shilong on 9/24/15.
//  Copyright (c) 2015 Agora. All rights reserved.
//

#import "AGSProfileViewController.h"
#import "AGSSelectMediaViewController.h"
#import "AGSAlertView.h"

@interface AGSProfileViewController ()

@property (weak, nonatomic) IBOutlet UITextField *key;
@property (weak, nonatomic) IBOutlet UITextField *username;

@property (weak, nonatomic) IBOutlet UIButton *modifyButton;

@end
@implementation AGSProfileViewController

- (IBAction)didClickModifyButton:(id)sender {
    
    if (self.key.text.length == 0) {
        UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:nil
                                                            message:NSLocalizedString(@"Please Enter Vendor Key", @"")
                                                           delegate:nil
                                                  cancelButtonTitle:NSLocalizedString(@"Done", @"")
                                                  otherButtonTitles:nil];
        [alertView show];
        return;
    }
    if (self.username.text.length == 0) {
        UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:nil
                                                            message:NSLocalizedString(@"Please Enter User Name", @"")
                                                           delegate:nil
                                                  cancelButtonTitle:NSLocalizedString(@"Done", @"")
                                                  otherButtonTitles:nil];
        [alertView show];
        return;
    }
    
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    [userDefaults setObject:self.key.text forKey:AGSKeyVendorKey];
    [userDefaults setObject:self.username.text forKey:AGSKeyUsername];
    
    [self.navigationController popViewControllerAnimated:YES];
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    //
    self.title = NSLocalizedString(@"Personal Info", @"");
    self.key.placeholder = NSLocalizedString(@"Enter Vendor Key", @"");
    self.username.placeholder = NSLocalizedString(@"User", @"");
    [self.modifyButton setTitle:NSLocalizedString(@"Update", @"") forState:UIControlStateNormal];
    
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    self.key.text = [userDefaults objectForKey:AGSKeyVendorKey];
    self.username.text = [userDefaults objectForKey:AGSKeyUsername];
}

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
    
    [self.key becomeFirstResponder];
}
@end

File: agora-easycall-ios-swift/SampleSwift/AGSProfileViewController.swift
//
//  AGSProfileViewController.swift
//  SampleSwift
//
//  Created by let on 10/29/15.
//  Copyright © 2015 Agora. All rights reserved.
//

import UIKit

class AGSProfileViewController: UIViewController {

    @IBOutlet weak var key: UITextField!
    @IBOutlet weak var username: UITextField!
    
    @IBOutlet weak var modifyButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        
        //
        title = NSLocalizedString("Personal Info", comment: "")
        key.placeholder = NSLocalizedString("Enter Vendor Key", comment: "")
        username.placeholder = NSLocalizedString("User", comment: "")
        modifyButton.setTitle(NSLocalizedString("Update", comment: ""), forState: UIControlState.Normal)
        
        let defaults = NSUserDefaults.standardUserDefaults()
        key.text = defaults.objectForKey(AGSKeys.vendorKey) as? String
        username.text = defaults.objectForKey(AGSKeys.username) as? String
    }
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
        
        key.becomeFirstResponder()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func didClickModifyButton(sender: AnyObject) {
        
        if key.text?.characters.count == 0 {
            let alert = UIAlertView(title: "", message: NSLocalizedString("Please Enter Vendor Key", comment: ""), delegate: nil, cancelButtonTitle: NSLocalizedString("Done", comment: ""))
            alert.show()
            return;
        }
        if username.text?.characters.count == 0 {
            let alert = UIAlertView(title: "", message: NSLocalizedString("Please Enter User Name", comment: ""), delegate: nil, cancelButtonTitle: NSLocalizedString("Done", comment: ""))
            alert.show()
            return;
        }
        
        let defaults = NSUserDefaults.standardUserDefaults()
        defaults.setObject(key.text, forKey: AGSKeys.vendorKey)
        defaults.setObject(username.text, forKey: AGSKeys.username)
        
        navigationController?.popViewControllerAnimated(true)
    }
}
